### PR TITLE
Set stricter CI runner and module versions

### PR DESCRIPTION
**Pull request by cristianrgreco**
- [x] Pin Podman version
~~- [ ] Pin Docker version~~
~~- [ ] Pin Colima version~~
- [x] Pin CI agent versions
- [x] Pin module versions

### CODE REVIEW

**Review by HofmeisterAn — APPROVED** [review]

File: .github/workflows/test-main.yml [hunk 1]
```diff
@@ -105,14 +105,14 @@ jobs:
...
-        os-version: [ macos-latest ]
+        os-version: [ macos-13 ]
         node-version: [ 16.x, 18.x, 20.x ]
     steps:
       - name: Code checkout
         uses: actions/checkout@v3
       - name: Setup Colima
         run: |
-          brew install docker docker-compose
+          brew install docker docker-compose colima
```
[the remark this answers is not in the document]
Comment (cristianrgreco) [reply .github/workflows/test-main.yml:110]: With brew it is _impossible_ :sob:

[the remark this answers is not in the document]
Comment (HofmeisterAn) [reply .github/workflows/test-main.yml:110]: > With brew it is _impossible_ 😭

Hm, why is that? Cannot you simply use `@0.5.4`?

[the remark this answers is not in the document]
Comment (cristianrgreco) [reply .github/workflows/test-main.yml:110]: That would be the dream, unfortunately not. Only some packages publish versioned packages. There _was_ a way to install specific versions by pointing to a URL (see attempt here https://github.com/testcontainers/testcontainers-node/pull/548/commits/a16d78c54364d2e792a64a69c86ec8033afaf3ba), however support for this was removed (https://github.com/Homebrew/brew/issues/8791)

[the remark this answers is not in the document]
Comment (HofmeisterAn) [reply .github/workflows/test-main.yml:110]: Thanks! I was not aware that this is specific to packages.

**Pull request merged**